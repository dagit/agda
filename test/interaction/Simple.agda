
module Simple where

data Nat : Set where
  z : Nat
  s : Nat → Nat

data Exp : Set where
  val : Nat → Exp
  throw : Exp

data Maybe (A : Set) : Set where
  Just    : A → Maybe A
  Nothing : Maybe A

eval : Exp → Maybe Nat
eval (val n)        = ?
eval throw          = {!!}
