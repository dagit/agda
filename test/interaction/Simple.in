top_command    (cmd_goals currentFile)
goal_command 0 (cmd_goal_type_context Normalised) ""
goal_command 0 cmd_refine "Just"
goal_command 2 cmd_give "n"
goal_command 1 cmd_give "s z"
goal_command 1 cmd_give "Nothing"
