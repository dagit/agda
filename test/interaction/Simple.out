agda2_mode_code (agda2-status-action "")
Checking Simple (Simple.agda).
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*All Goals*"
                 "?0 : Maybe Nat
?1 : Maybe Nat
")
agda2_mode_code (setq agda2-response '(0 1))
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*Goal type etc.*"
                 "n : Nat
————————————————————————————————————————————————————————————
Goal: Maybe Nat")
agda2_mode_code (agda2-give-action 0 "Just ?")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*All Goals*"
                 "?1 : Maybe Nat
?2 : Nat
")
agda2_mode_code (agda2-give-action 2 'no-paren)
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action "*All Goals*" "?1 : Maybe Nat
")
agda2_mode_code (agda2-info-action
                 "*Error*"
                 "1,1-4
Nat !=< Maybe Nat of type Set
when checking that the expression s z has type Maybe Nat")
agda2_mode_code (agda2-status-action "")
*** Exception: ExitFailure 1
agda2_mode_code (agda2-give-action 1 'no-paren)
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action "*All Goals*" "")
